jwst/tso_photometry/tso_photometry.py

```diff
@@ -79,25 +79,28 @@ def tso_aperture_photometry(datamodel, xcenter, ycenter, radius, radius_inner,
                 'subarray.  No background subtraction was performed.')
 
         for i in np.arange(nimg):
-            aperture_sum.append(np.sum(datamodel.data[i, :, :]))
+            aperture_sum.append(np.nansum(datamodel.data[i, :, :]))
             aperture_sum_err.append(
-                np.sqrt(np.sum(datamodel.err[i, :, :]**2)))
+                np.sqrt(np.nansum(datamodel.err[i, :, :]**2)))
     else:
         info = ('Photometry measured in a circular aperture of r={0} '
                 'pixels.  Background calculated as the mean in a '
                 'circular annulus with r_inner={1} pixels and '
                 'r_outer={2} pixels.'.format(radius, radius_inner,
                                              radius_outer))
         for i in np.arange(nimg):
-            aper_sum, aper_sum_err = phot_aper.do_photometry(
-                datamodel.data[i, :, :], error=datamodel.err[i, :, :])
-            ann_sum, ann_sum_err = bkg_aper.do_photometry(
-                datamodel.data[i, :, :], error=datamodel.err[i, :, :])
-
-            aperture_sum.append(aper_sum[0])
-            aperture_sum_err.append(aper_sum_err[0])
-            annulus_sum.append(ann_sum[0])
-            annulus_sum_err.append(ann_sum_err[0])
+            aperstats = ApertureStats(datamodel.data[i, :, :],
+                                      phot_aper,
+                                      error=datamodel.err[i, :, :])
+
+            annstats = ApertureStats(datamodel.data[i, :, :],
+                                     bkg_aper,
+                                     error=datamodel.err[i, :, :])
+
+            aperture_sum.append(aperstats.sum)
+            aperture_sum_err.append(aperstats.sum_err)
+            annulus_sum.append(annstats.sum)
+            annulus_sum_err.append(annstats.sum_err)
```

Comment:
Is there a reason to switch to ApertureStats here instead of just supplying a mask to do_photometry?

Reply:
Not really.  I could switch to using a mask if that is preferred. I just went with Ian Wong's suggestion in the original ticket because he had more experience than I did with photutils.

Reply:
Oh, I see that comment now.  It sounds like there is a good reason to switch!  Thanks!